### PR TITLE
Add api/nonprofits/:nonprofit_id/supporters and api/nonprofits/:nonprofit_id/supporters/:id

### PR DESCRIPTION
**Pull request by wwahammy**
Depends on #639

### CODE REVIEW

File: spec/requests/api/supporters_request_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+			sign_in user
+			get "/api/nonprofits/#{nonprofit.id}/supporters/#{supporter.id}"
+			expect(response).to have_http_status(:success)
+		end
+
+		it 'returns http success when not logged in' do
+			get "/api/nonprofits/#{nonprofit.id}/supporters/#{supporter.id}"
+			expect(response).to have_http_status(:unauthorized)
+		end
+	end
```
Comment (clarissalimab) [inline spec/requests/api/supporters_request_spec.rb:43]: Wouldn't it be a good idea to test how the return looks like? I think it also helps as a documentation for the API.

**Review by clarissalimab — COMMENTED** [review]
Just a :memo:

File: spec/requests/api/supporters_request_spec.rb (hunk 1 above)
Comment (wwahammy) [reply spec/requests/api/supporters_request_spec.rb:43]: Probably.

There's multiple schools of thought in the Rails community on that but it wouldn't hurt to do that. :+1:

Comment (wwahammy) [reply spec/requests/api/supporters_request_spec.rb]: I've added this to the specs!

**Pull request merged**